controllers/application_controller.go

```diff
@@ -58,8 +59,9 @@ func (r *ApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, nil
 	}
 
-	resources, err := r.updateComponents(ctx, &app)
-	newApplicationStatus := r.getNewApplicationStatus(ctx, &app, resources, err)
+	var errs []error
+	resources := r.updateComponents(ctx, &app, &errs)
```

Comment:
suggest getting errs list from updateComponents instead of passing it and updating it.
the var err []error can be defined inside the function and returned here.